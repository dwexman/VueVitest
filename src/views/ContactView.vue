<template>
  <div class="contact-container">
    <h1>Contact Page</h1>
    <p>This is the Contact page!</p>
    <router-link to="/" class="back-link">Back to Home</router-link>
  </div>
</template>

<script>
export default {
  name: 'ContactView',
};
</script>

<style>
.contact-container {
  text-align: center;
  padding: 20px;
}

.back-link {
  display: inline-block;
  margin-top: 20px;
  padding: 10px 20px;
  background-color: #ffc107;
  color: black;
  text-decoration: none;
  border-radius: 5px;
  transition: background-color 0.3s ease;
}

.back-link:hover {
  background-color: #e0a800;
}
</style>
