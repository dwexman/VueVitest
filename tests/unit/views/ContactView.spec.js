import { mount } from '@vue/test-utils';
import ContactView from '../../../src/views/ContactView.vue';

describe('ContactView.vue', () => {
  it('renders the Contact page', () => {
    const wrapper = mount(ContactView);
    expect(wrapper.text()).toContain('Contact Page');
    expect(wrapper.text()).toContain('This is the Contact page!');
  });

  it('matches the snapshot', () => {
    const wrapper = mount(ContactView);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
