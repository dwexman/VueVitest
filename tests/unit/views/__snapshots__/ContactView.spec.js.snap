// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ContactView.vue > matches the snapshot 1`] = `
"<div class="contact-container">
  <h1>Contact Page</h1>
  <p>This is the Contact page!</p>
  <router-link to="/" class="back-link">Back to Home</router-link>
</div>"
`;
